### DISCUSSION TITLE
LEDドライバなにを使おう？

### DISCUSSION THREAD

**Discussion by matsujirushi** [algyan/XIAOGYAN#1] [General]
HT16K33が良さそうなんだけど、

* 5V（3.3Vで動くみたいではあるが、データシート上はNG）
* 入手し難い（秋月電子通商で欠品）

なので、、、どうしようかな。

**Comment by tatsu1225** [comment]
試した事はないのですが7seg用のLEDドライバを転用する作例があるみたいです．
https://nullnuma.hatenablog.com/entry/2018_08_12_tm1640

TM1640ってなドライバICですと，80円/個から秋月で販売されているようですけど…5V動作品でした(^q^
https://akizukidenshi.com/catalog/g/gI-13225/

あとは，595を2段つなげて使うパターンとかですかね？
https://akizukidenshi.com/catalog/g/gI-14053/

**Comment by matsujirushi** [comment]
情報ありがとうございます！！

# TM1640

5Vがアレですが、それ以外は良さそう。

* 8x8 LEDマトリックスでも使えそう
* 入手OK
* 通信がI2Cではなく、CLKとDIN。
* 5V

# HC565

シフトレジスタですね。
安心安全なのはこっちだけど、LED駆動専用品じゃないから、電流制御が無いのがちょっとなーと思っています。

**Comment by y-ohe** [comment]
LEDドライバの方じゃないんですが、TM1640やHT16K33だと、16comx8segまで駆動できるので、「2色ドットマトリクスLED(赤緑) 8x8ドット LTP-12188M-08 [I-09878]」を使うのはどうでしょうか？

**Reply by matsujirushi** [reply]
ナイスアドバイス！！
良いですね！
2色のものにしましょう。

**Comment by matsujirushi** [comment]
TM1640を入手して、動作確認しました。

* 5V電源、3.3V通信 ... 問題なく動作
* 3.3V電源、3.3V通信 ... 問題なく動作

ディスコン品なのが気になりますが、現時点、秋月に在庫沢山あるので、**TM1640**を採用します！
